src/tools/perf/lib/libperf.c

```diff
@@ -109,54 +109,54 @@ static const char *perf_atomic_fop[] = {
  *  Cambridge University Press, 1992, Section 8.5, ISBN 0-521-43108-5
  *  This code by Nicolas Devillard - 1998. Public domain.
  */
-static ucs_time_t __find_median_quick_select(ucs_time_t arr[], int n)
+static ucs_time_t __find_percentile_quick_select(ucs_time_t arr[], int n, double rank)
 {
-    int low, high ;
-    int median;
+    int low, high;
+    int percentile_idx;
     int middle, ll, hh;
 
 #define ELEM_SWAP(a,b) { register ucs_time_t t=(a);(a)=(b);(b)=t; }
 
-    low = 0 ; high = n-1 ; median = (low + high) / 2;
+    low = 0; high = n - 1; percentile_idx = high * (rank / 100.0);
     for (;;) {
         if (high <= low) /* One element only */
-            return arr[median] ;
+            return arr[percentile_idx];
```

Comment:
more-of-a-comment: not a fan of uncuddled "if"s, but I don't see anything in the code style about it

Reply:
Neither am I, will fix.